frontend/src/components/UI/select/index.tsx

```diff
@@ -1,51 +1,114 @@
-import { FieldValues, UseFormRegister } from 'react-hook-form';
-import { RiArrowDownSFill } from 'react-icons/ri';
+import { useEffect, useRef, useState } from 'react';
+import {
+  FieldValues,
+  RegisterOptions,
+  UseFormRegister,
+  UseFormSetValue,
+} from 'react-hook-form';
+import { RiArrowDownSFill, RiArrowUpSFill } from 'react-icons/ri';
+import { ShelvesOptionTypes } from '../../modal/modal.interfaces';
 import { Error } from '../../text/text-styles';
-import { SelectContainer, StyledSelect } from './select-styles';
+import {
+  DropDownContainer,
+  DropDownHeader,
+  DropDownList,
+  DropdownItem,
+  DropdownListContainer,
+} from './select-styles';
 
 interface SelectProps {
   optionsData: { value: string; text: string }[];
   selectName: string;
   placeHolder: string;
   register: UseFormRegister<FieldValues>;
   error: { message: string };
-  validation?: {
-    validate: (value: string) => true | 'Cant select same shelfs';
-  };
+  validation?: RegisterOptions;
+  setValue?: UseFormSetValue<FieldValues>;
+  variant?: string;
```

Comment:
Maybe we can have string values of possible variants as type? 
E.g. "primary" | "secondary"